### PR TITLE
Token management

### PR DESCRIPTION
**Pull request by jennybc**
I can't get you completely back in business but I think these are good changes. We'll need to do a few other things.

### CODE REVIEW

File: tests/testthat/helper.R [hunk 1]
```diff
@@ -0,0 +1,8 @@
+if (gargle:::secret_can_decrypt("gcalendr")) {
+  json <- gargle:::secret_read("gcalendr", "gcalendr-testing.json")
```
Comment (jennybc) [inline tests/testthat/helper.R:2]: We need to actually put an encrypted service account token in the expected location for this to work. Namely at `inst/secret/gcalendr-testing.json`. We will do that like so:

```
gargle:::secret_write(
  package = "gcalendr",
  name = "gcalendr-testing.json",
  input = "a/very/private/local/folder/gcalendr-testing.json"
)
```

after we download a service account token to `a/very/private/local/folder/gcalendr-testing.json`.

Excerpting the important bits of https://gargle.r-lib.org/articles/articles/managing-tokens-securely.html

File: tests/testthat/test-gcalendr.R [hunk 2]
```diff
@@ -1,64 +1,57 @@
-context("Read calendar events")
```
Comment (jennybc) [inline tests/testthat/test-gcalendr.R:1]: `context()`s are no longer needed and are semi-discouraged. I left them in the other test files but got rid of this one.

File: tests/testthat/test-gcalendr.R [hunk 3]
```diff
@@ -1,64 +1,57 @@
...
+test_that("Can read events", {
+  skip_if_no_token()
```
Comment (jennybc) [inline tests/testthat/test-gcalendr.R:2]: This is the nicest way to gracefully skip over token-needing tests when no token is available.

File: tests/testthat/test-gcalendr.R [hunk 4]
```diff
@@ -1,64 +1,57 @@
...
+  expect_equal(ncol(calendar_ids), 3)
 
-    secret <- gargle:::secret_read("gcalendr", "gmail-token")
-    token <- gargle::credentials_service_account(path = rawToChar(secret))
-    calendar_auth("apdevries@gmail.com", path = token, use_oob = TRUE)
 
-    expect_true(calendar_has_token())
+  # JENNY QUESTION: I'm worried about this. The test should expect to be
+  # logged in as the service account. But maybe I just don't understand what
+  # this function does.
```
Comment (jennybc) [inline tests/testthat/test-gcalendr.R:11]: We should talk about this. You might need to auth as the service account and create some events, so we know what we expect to find here?

In googlesheets4, I'm putting the code that creates my test fixtures (Google Sheets in my case) here:

https://github.com/tidyverse/googlesheets4/tree/master/tools/test-fixtures

It's nice to have this programmatically recorded.

File: tests/testthat/test-gcalendr.R (hunk 2 above)
Comment (andrie) [reply tests/testthat/test-gcalendr.R:1]: Thanks!  I didn't know this.

File: tests/testthat/helper.R (hunk 1 above)
Comment (andrie) [reply tests/testthat/helper.R:2]: This is helpful.  I've renamed my secret file to follow the `package-testing.json` convention and this *is* simplifying things!

**Pull request merged**